errors_test.go

```diff
@@ -0,0 +1,536 @@
+		{name: "Conflict (with messagef)", args: args{err: NewConflictMessagef("msg %v", "arg")}, want: true},
+		{name: "Conflict (with message and error)", args: args{err: NewConflictMessageWithError(stderr.New("err"), "msg")}, want: true},
+		{name: "Conflict (with message and errorf)", args: args{err: NewConflictMessageWithErrorf(stderr.New("err"), "msg %v", "arg")}, want: true},
+
+		{name: "InternalServerError", args: args{err: NewInternalServerError()}, want: true},
+		{name: "InternalServerError (with error)", args: args{err: NewInternalServerErrorWithError(stderr.New("msg"))}, want: true},
+		{name: "InternalServerError (with message)", args: args{err: NewInternalServerErrorMessage("msg")}, want: true},
+		{name: "InternalServerError (with messagef)", args: args{err: NewInternalServerErrorMessagef("msg %v", "arg")}, want: true},
+		{name: "InternalServerError (with message and error)", args: args{err: NewInternalServerErrorMessageWithError(stderr.New("err"), "msg")}, want: true},
+		{name: "InternalServerError (with message and errorf)", args: args{err: NewInternalServerErrorMessageWithErrorf(stderr.New("err"), "msg %v", "arg")}, want: true},
+
+		{name: "ServiceUnavailable", args: args{err: NewServiceUnavailable()}, want: true},
+		{name: "ServiceUnavailable (with error)", args: args{err: NewServiceUnavailableWithError(stderr.New("msg"))}, want: true},
+		{name: "ServiceUnavailable (with message)", args: args{err: NewServiceUnavailableMessage("msg")}, want: true},
+		{name: "ServiceUnavailable (with messagef)", args: args{err: NewServiceUnavailableMessagef("msg %v", "arg")}, want: true},
+		{name: "ServiceUnavailable (with message and error)", args: args{err: NewServiceUnavailableMessageWithError(stderr.New("err"), "msg")}, want: true},
+		{name: "ServiceUnavailable (with message and errorf)", args: args{err: NewServiceUnavailableMessageWithErrorf(stderr.New("err"), "msg %v", "arg")}, want: true},
+
+		{name: "Other error", args: args{err: stderr.New("err")}, want: false},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			if got := HasClientMessage(tt.args.err); got != tt.want {
+				t.Errorf("HasClientMessage() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestHasBadRequest(t *testing.T) {
+	t.Parallel()
+
+	type args struct {
+		err error
+	}
+	tests := []struct {
+		name string
+		args args
+		want bool
+	}{
+		{name: "BadRequest (with message)", args: args{err: NewBadRequestMessage("msg")}, want: true},
+		{name: "BadRequest (with messagef)", args: args{err: NewBadRequestMessagef("msg %v", "arg")}, want: true},
+		{name: "BadRequest (with message and error)", args: args{err: NewBadRequestMessageWithError(stderr.New("err"), "msg")}, want: true},
+		{name: "BadRequest (with message and errorf)", args: args{err: NewBadRequestMessageWithErrorf(stderr.New("err"), "msg %v", "arg")}, want: true},
+		{name: "Other error", args: args{err: stderr.New("err")}, want: false},
```

Comment:
```suggestion
		{name: "BadRequest", args: args{err: NewBadRequest()}, want: true},
		{name: "BadRequest (with error)", args: args{err: NewBadRequestWithError(stderr.New("msg"))}, want: true},
		{name: "BadRequest (with message)", args: args{err: NewBadRequestMessage("msg")}, want: true},
		{name: "BadRequest (with messagef)", args: args{err: NewBadRequestMessagef("msg %v", "arg")}, want: true},
		{name: "BadRequest (with message and error)", args: args{err: NewBadRequestMessageWithError(stderr.New("err"), "msg")}, want: true},
		{name: "BadRequest (with message and errorf)", args: args{err: NewBadRequestMessageWithErrorf(stderr.New("err"), "msg %v", "arg")}, want: true},
		{name: "Other error", args: args{err: stderr.New("err")}, want: false},
```

Also, the same comment applies to the other similar ones.